libs/motor/inc/public/tfc/motors/mock_motor.hpp

```diff
@@ -0,0 +1,55 @@
+#include <string>
+
+#include <mp-units/format.h>
+#include <mp-units/ostream.h>
+#include <mp-units/systems/isq/isq.h>
+#include <mp-units/systems/si/si.h>
+
+#include <tfc/confman.hpp>
+#include <tfc/confman/observable.hpp>
+#include <tfc/motors/errors.hpp>
+#include <tfc/motors/impl.hpp>
+#include <tfc/utils/units_glaze_meta.hpp>
+
+namespace tfc::motor::types {
+using mp_units::QuantityOf;
+using tfc::confman::observable;
+using namespace mp_units::si::unit_symbols;
+
+/**
+ * @brief mock motor class
+ * @details This struct is only used for testing purposes.
+ * none of the methods should be called in production.
+ * This struct is also simple and does not rely on time by
+ * default.
+ */
+
+template <typename clock_t>
+struct mock_motor {
+  explicit virtual_motor(boost::asio::io_context&) {}
```

Comment:
discrepancy here, does it compile?

Reply:
This idea is not included anywhere i believe. Lets revisit when there are some tests written on top of this and in usage.